### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by noam-mendelson**
This pull request contains my implementation of a Convolutional Vision Transformer (CViT) for classification of Alzheimer’s Disease using the ADNI dataset (Task 6). Files included are:
-       dataset.py: data loader for loading and pre-processing data
-       modules.py: source code of the CViT components
-        train.py: source code for training, validating and saving model
-       predict.py: example usage of trained model
-       ./Images: images detailing results of testing, and supporting images for
-       README.md: thorough documentation of project
-       Models.pth: trained model weights saved
 
Thanks, 
Noam Mendelson (47432923)

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent

Commit messages: Excellent, detailed.

Code:
- Uses Covolutional Vision transformer
- Well commented
- Good model design

Functionality/Performance:
- Val accuracy is 67.8%
- **Don't have train, val and test sets. Only have train and val sets.**
- Did patient-level split
- Preprocssing data correctly
- Performed data augmentation
- Well done

General comments.
Good choice on CViT, should be helpful to deal with small dataset and overfitting problem. Can try more hyperparameter tuning and make the model more smaller. Also seems like your model can be training longer. It's a little bit underfitting just from the training and val loss plot.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present, didnt use own folder and committed to trunk of repo -2
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, data splits -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove model files and move code into own folder as required -2
Request Description good

**Comment by shakes76** [comment]
Can't merge into repo because of large model file, but Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by noam-mendelson** [comment]
I am working on feedback- I am currently in hospital so it is difficult for me to do so.

**Comment by wangzhaomxy** [comment]
> I am working on feedback- I am currently in hospital so it is difficult for me to do so.

I just marked all problem pull requests and was making sure I had covered everyone. You can still get marks after finishing the adjustment. I will get another inspection before the due date. However, we cannot wait for endless time, we have to update your grades to the school in a few days. Please adjust it ASAP.
